### DISCUSSION TITLE
Apollo Sandbox not working in offline environment

### DISCUSSION THREAD

**Discussion by Achder** [apollographql/apollo-server#7833] [Q&A]
Hi,

the company I'm working on develops exclusively in remote environments with tight restriction on outside communication. The only access to the outside for me so far is through an NPM proxy.
In this scenario, the Apollo Sandbox is not able to load because it tries to fetch data from a CDN.
Is there a way to make it run local only?

**Comment by david-hanger** [comment]
I'll add a bump to this.  My team works primarily on air-gapped networks, and we have been entirely unable to get Sandbox to work after some weeks of trying (including an attempt at mirroring the entire CDN and forking the repo(s) to rewrite all the hostnames).  We have Playground working, but the features of Sandbox would be a huge boon.  Unfortunately our schema is not allowed to be brought down to a lower security level for internet access to use Sandbox.

**Comment by Achder** [comment]
I have it working with graphiql: https://github.com/graphql/graphiql
It can be configured to run without internet access
